cmd/clusterctl/examples/vsphere/machines.yaml.template

```diff
@@ -14,29 +14,21 @@ items:
           apiVersion: "vsphereproviderconfig/v1alpha1"
           kind: "VsphereMachineProviderConfig"
           machineSpec:
-            datacenter: ""
-            datastore: ""
-            resourcePool: ""
+            datacenter: "${VSPHERE_DATACENTER}"
+            datastore: "${VSPHERE_DATASTORE}"
+            resourcePool: "${VSPHERE_RESOURCE_POOL}"
+            vmFolder: "${VSPHERE_FOLDER}"
             networks:
-            - networkName: ""
+            - networkName: "${VSPHERE_NETWORK}"
               ipConfig:
                 networkType: static
```

Comment:
Hi @andrewsykim,

I think the networkType should default to dhcp, right? Otherwise we should force the selection of IP address, subnet, gateway information, and possibly DNS servers.